orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -426,7 +452,7 @@ def _get_inference_graph(self, *inputs, **kwargs):
 
         # Setup dynamic axes for onnx model
         input_names, dynamic_axes, self._input_names_require_grad = _parse_inputs_for_onnx_export(self._original_module, *inputs, **kwargs)
-        output_names, output_dynamic_axes = _parse_outputs_for_onnx_export(self._original_module, inputs)
+        output_names, output_dynamic_axes, self._sample_user_output = _parse_outputs_for_onnx_export(self._original_module, inputs)
```

Comment:
Could `self._sample_user_output` contain large tensors here? Unless I'm mistaken I *think* all we need to keep is `type(sample_user_output)`.

Reply:
You're right. I had initially started off with just saving the type. Later, I was experimenting with having more complex return types (such as dictionary of dictionaries ...), which was when I changed this to save the entire sample output.
But I think since this PR only addresses the simple types, saving just the type is enough. Will update the PR with the change now while I think of ways to save types of more complex outputs.